validator_client/src/preparation_service.rs

```diff
@@ -187,68 +188,74 @@ impl<T: SlotClock + 'static, E: EthSpec> PreparationService<T, E> {
         Ok(())
     }
 
-    fn collect_preparation_data(&self, spec: &ChainSpec) -> Vec<ProposerPreparationData> {
-        let log = self.context.log();
-
-        let fee_recipient_file = self
-            .fee_recipient_file
+    pub fn read_fee_recipient_file(&self) -> Option<FeeRecipientFile> {
+        self.fee_recipient_file
             .clone()
             .map(|mut fee_recipient_file| {
                 fee_recipient_file
                     .read_fee_recipient_file()
                     .map_err(|e| {
                         error!(
-                            log,
+                            self.context.log(),
                             "Error loading fee-recipient file";
                             "error" => ?e
                         );
                     })
                     .unwrap_or(());
                 fee_recipient_file
-            });
+            })
+    }
+
```

Comment:
So the logic for retrieving the fee recipient for a given public key was pulled out into `get_preparation_data()` for use in the API call. Because this function is called in a loop inside `collect_preparation_data`, I made it accept a `FeeRecipientFile` object instead of reading from disk so that doesn't need to happen for each iteration of the loop. This required me to create the `read_fee_recipient_file` function above